src/test/streams_tests.cpp

```diff
@@ -403,6 +448,17 @@ BOOST_AUTO_TEST_CASE(streams_buffered_file_rand)
                 break;
             }
             case 3: {
+                // Skip is similar to the "read" cases above, except
+                // we don't receive the data.
+                size_t nSize = InsecureRandRange(5);
+                if (currentPos + nSize > fileSize)
+                    continue;
+                bf.SetLimit(currentPos + nSize);
+                bf.Skip(nSize);
+                currentPos += nSize;
+                break;
```

Comment:
I agree that looks strange, but this case isn't an isolated test, we're within a 100-iteration loop. As the test continues to loop, any `Skip()` bug would cause the `BOOST_CHECK_EQUAL` at line 407 to fail (and also the other cases to fail since `currentPos` keeps track of where we should be in the stream).